src/coreclr/jit/inlinepolicy.cpp

```diff
@@ -488,8 +488,13 @@ bool DefaultPolicy::BudgetCheck() const
     // The strategy tracks the amout of inlining done so far,
     // so it performs the actual check.
     //
-    InlineStrategy* strategy   = m_RootCompiler->m_inlineStrategy;
-    const bool      overBudget = strategy->BudgetCheck(EstimatedTotalILSize());
+    InlineStrategy* strategy = m_RootCompiler->m_inlineStrategy;
+
+    const bool preferSpeed = m_RootCompiler->opts.jitFlags->IsSet(JitFlags::JIT_FLAG_SPEED_OPT);
+    const bool isAot       = m_RootCompiler->opts.IsReadyToRun();
+
+    // Ignore time budget for AOT + PreferSpeed, otherwise check if we run out of it.
+    const bool overBudget = !(isAot && preferSpeed) && strategy->BudgetCheck(EstimatedTotalILSize());
```

Comment:
Shouldn't crossgen2/R2R have the same behavior here as NativeAOT?

Reply:
It is the same currently, but AFAIK `--Ot` can not be easily passed to crossgen2 which is used through Msbuild task in 99% so that's why I used "click bait" NativeAOT title despite the fact it's applied for both 🙂